package.json

```diff
@@ -70,6 +70,7 @@
     "postbuild:docs": "apidoc -i doc/ -o doc/",
     "build:flow": "node scripts/generate-flow-types",
     "build:ts": "node scripts/generate-typescript-types",
+    "validate:dts": "npx --package typescript tsc index.d.ts",
```

Comment:
For node 9, I think it's a safe bet that npx is installed.

Reply:
I'm not sure if there will be a speed difference, but to take advantage of module caching on Travis, it might be worth adding `npm install -g typescript` to the `install` script.

Reply:
I don't have too strong of an opinion TBH - I agree with where you're coming from, only difference is that this is a one-liner

Reply:
I would add `typescript` as dev dependency and the set the task to `"validate:ts": "typescript tsc index.d.ts"`. I would also add `"postbuild:ts": "npm run -s validate:ts"`.

Why `validate:dts` and not `validate:ts`? Is that a common convention? Should we rename `build:ts` to `build:dts`, too?

Reply:
The created file is a `.d.ts` (a typescript definition file) so I figured I was just validating that the definition file passes the compiler's linter. Happy to add typescript as a dep etc 👍

Reply:
For simplicities sake I just made them both `ts`